### PR TITLE
feat: rate limit worker

### PR DESCRIPTION
**Pull request by assimovt**
fixes FTD-2074, FTD-2142

### CODE REVIEW

File: packages/service-utils/src/core/authorize/types.ts [hunk 1]
```diff
@@ -12,3 +12,13 @@ export type AuthorizationResult =
       errorMessage: string;
       errorCode: string;
     };
+
+export type RateLimitResult =
```
Comment (arcoraven) [inline packages/service-utils/src/core/authorize/types.ts:16]: This type is repeated below in rateLimit/types.ts
We don't need this one here right?

**Review by arcoraven — COMMENTED** [review]
Overall logic looks good 👍 
You wrote the rate limiter in a way that seems fairly straightfoward to unit test.

File: packages/service-utils/src/core/rateLimit/index.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      rateLimited: false,
+    };
+  }
+
+  // round current time to the nearest DEFAULT_RATE_LIMIT_WINDOW_SECONDS
+  const bucketId =
+    Math.floor(Date.now() / (1000 * DEFAULT_RATE_LIMIT_WINDOW_SECONDS)) *
+    DEFAULT_RATE_LIMIT_WINDOW_SECONDS;
+  const key = [serviceScope, accountId, bucketId].join(":");
+  const value = parseInt((await cacheOptions.get(key)) || "0");
```
[the remark this answers is not in the document]
Comment (arcoraven) [reply packages/service-utils/src/core/rateLimit/index.ts:33]: Nvm, I see typescript complains when I try this

**Comment by arcoraven** [comment]
Added some unit tests (mostly written by ChatGPT)

File: packages/service-utils/src/core/api.ts [hunk 3]
```diff
@@ -1,8 +1,7 @@
 import type { ServiceName } from "./services";
 
-
 export type CoreServiceConfig = {
-  enforceAuth: boolean;
+  enforceAuth?: boolean;
```
[the remark this answers is not in the document]
Comment (assimovt) [reply packages/service-utils/src/core/api.ts:4]: yeah cause we're reusing CoreServiceConfig in rateLimit and this is not required. it always defaults to false.

File: packages/service-utils/src/core/usageLimit/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    };
+  }
+
+  if (
+    serviceScope === "storage" &&
+    (usage.storage?.sumFileSizeBytes || 0) > limit
+  ) {
+    return {
+      usageLimited: true,
+      status: 402,
```
Comment (arcoraven) [inline packages/service-utils/src/core/usageLimit/index.ts:27]: Interesting, I didn't know 402 existed but I see it's [experimental](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/402). No strong opinion. [403 Forbidden](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) would be a reasonable alternative.

**Review by arcoraven — APPROVED** [review]

**Review by arcoraven — APPROVED** [review]

**Pull request merged**